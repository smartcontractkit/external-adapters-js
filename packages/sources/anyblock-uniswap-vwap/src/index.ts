import { expose } from '@chainlink/ea-bootstrap'
import { execute } from './adapter'

const NAME = 'ANYBLOCK_UNISWAP_VWAP'

export = { NAME, execute, ...expose(execute) }
