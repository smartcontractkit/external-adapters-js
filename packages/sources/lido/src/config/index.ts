import { Requester, util } from '@chainlink/ea-bootstrap'
import type { Config } from '@chainlink/ea-bootstrap'

export const NAME = 'LIDO'
export const ENV_POLYGON_RPC_URL = 'POLYGON_RPC_URL'
export const ENV_POLYGON_CHAIN_ID = 'POLYGON_CHAIN_ID'
export const DEFAULT_CHAIN_ID = '137'
export const MATIC_AGGREGATOR_PROXY = '0xAB594600376Ec9fD91F8e885dADF0CE036862dE0'
export const STMATIC_RATE_PROVIDER = '0xdEd6C522d803E35f65318a9a4d7333a22d582199'
export const DEFAULT_ENDPOINT = 'stmatic'

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix)
  config.api.baseURL ||= util.getEnv(ENV_POLYGON_RPC_URL) || ''
  config.chainId =
    parseInt(util.getEnv(ENV_POLYGON_CHAIN_ID) || DEFAULT_CHAIN_ID) ||
    util.getEnv(ENV_POLYGON_CHAIN_ID)
  config.defaultEndpoint = DEFAULT_ENDPOINT
  return config
}
