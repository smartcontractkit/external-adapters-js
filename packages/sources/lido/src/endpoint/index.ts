import type { TInputParameters as StmaticInputParameters } from './stmatic'

export type TInputParameters = StmaticInputParameters

export * as stmatic from './stmatic'
