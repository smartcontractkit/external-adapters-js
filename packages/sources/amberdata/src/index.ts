import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './adapter'
import { makeConfig, WSHandlerFactory, NAME } from './config'

export = { NAME, makeExecute, makeConfig, ...expose(makeExecute(), WSHandlerFactory()) }
