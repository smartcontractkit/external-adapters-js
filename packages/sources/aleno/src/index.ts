import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
import { Adapter } from '@chainlink/external-adapter-framework/adapter'
import { config } from './config'
import { price } from './endpoint'

export const adapter = new Adapter({
  defaultEndpoint: price.name,
  name: 'ALENO',
  config,
  endpoints: [price],
  rateLimiting: {
    tiers: {
      default: {
        rateLimit1m: 30,
      },
    },
  },
})

export const server = (): Promise<ServerInstance | undefined> => expose(adapter)
