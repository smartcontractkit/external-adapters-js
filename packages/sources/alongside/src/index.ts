import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
import { Adapter } from '@chainlink/external-adapter-framework/adapter'
import { collateral } from './endpoint'
import { config } from './config'

export const adapter = new Adapter({
  defaultEndpoint: collateral.name,
  name: 'ALONGSIDE',
  config,
  endpoints: [collateral],
})

export const server = (): Promise<ServerInstance | undefined> => expose(adapter)
