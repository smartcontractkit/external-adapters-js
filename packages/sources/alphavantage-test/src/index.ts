import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
import { PriceAdapter } from '@chainlink/external-adapter-framework/adapter'
import { config } from './config'
import { forex } from './endpoint'

export const adapter = new PriceAdapter({
  defaultEndpoint: forex.name,
  name: 'ALPHAVANTAGE',
  config,
  endpoints: [forex],
  rateLimiting: {
    tiers: {
      free: {
        rateLimit1m: 5,
        rateLimit1h: 20.83,
      },
      '49.99': {
        rateLimit1m: 75,
      },
      '99.99': {
        rateLimit1m: 150,
      },
      '149.99': {
        rateLimit1m: 300,
      },
      '199.99': {
        rateLimit1m: 600,
      },
      '249.99': {
        rateLimit1m: 1200,
      },
    },
  },
})

export const server = (): Promise<ServerInstance | undefined> => expose(adapter)
