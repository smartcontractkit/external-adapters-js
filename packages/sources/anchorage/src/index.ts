import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
import { config } from './config'
import { wallet } from './endpoint'
import { PoRAdapter } from '@chainlink/external-adapter-framework/adapter/por'

export const adapter = new PoRAdapter({
  defaultEndpoint: wallet.name,
  name: 'ANCHORAGE',
  config,
  endpoints: [wallet],
  rateLimiting: {
    tiers: {
      default: {
        rateLimit1m: 30,
        note: 'Docs: 10 requests per second per Organization, bursts of up to 100 requests per second',
      },
    },
  },
})

export const server = (): Promise<ServerInstance | undefined> => expose(adapter)
export const walletParameters = wallet.inputParameters
