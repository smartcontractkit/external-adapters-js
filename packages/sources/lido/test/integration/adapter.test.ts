import { AdapterRequest } from '@chainlink/ea-bootstrap'
import process from 'process'
import { server as startServer } from '../../src'
import { mockStmaticSuccess } from './fixtures'
import { setupExternalAdapterTest } from '@chainlink/ea-test-helpers'
import type { SuiteContext } from '@chainlink/ea-test-helpers'
import { SuperTest, Test } from 'supertest'

describe('execute', () => {
  const context: SuiteContext = {
    req: null,
    server: startServer,
  }

  const envVariables = {
    POLYGON_RPC_URL: process.env.POLYGON_RPC_URL || 'https://test-rpc-url/',
  }

  setupExternalAdapterTest(envVariables, context)
  describe('stmatic endpoint', () => {
    const data: AdapterRequest = {
      id: '1',
      data: {},
    }

    it('should return success', async () => {
      mockStmaticSuccess()

      const response = await (context.req as SuperTest<Test>)
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })
})
