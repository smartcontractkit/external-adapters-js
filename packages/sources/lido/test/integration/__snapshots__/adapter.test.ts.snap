// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute stmatic endpoint should return success 1`] = `
{
  "data": {
    "result": 69357317,
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": 69357317,
  "statusCode": 200,
}
`;
